tests/net/lib/coap/src/main.c

```diff
@@ -855,6 +906,23 @@ static int test_observer_server(void)
 		goto done;
 	}
 
+	payloadfrag = coap_packet_get_payload(request, &offset, &len);
+	if (!payloadfrag && len == 0xffff) {
+		NET_ERR("Invalid payload");
+		return -EINVAL;
+	} else if (!payloadfrag && !len) {
+		NET_INFO("Packet without payload\n");
+		goto done;
+	}
+
+	payloadfrag->data = payloadfrag->data + offset;
```

Comment:
why are you overwriting payloadfrag->data pointer ?
Can you try directly strcmp(payloadfrag->data + offset, "payload")

Reply:
Yes I can do in this way also.

Reply:
done